Builder/HtmlBuilder.php

```diff
@@ -14,267 +13,50 @@
-                    $attributes['href'] = $data['url'];
-                }
-
-                if ($data['target'] && '_self' !== $data['target']) {
-                    $attributes['target'] = $data['target'];
-                }
-
-                if ($data['nofollow'] && true === $data['nofollow']) {
-                    $attributes['rel'] = 'nofollow';
-                }
-
-                return $this->openNode(self::ENTITY_NODE, $attributes);
-            default:
-                throw new DraftjsException(sprintf('Unsupported entity type %s', $type));
-        }
-    }
-
-    /**
-     * @return string
-     */
-    private function closeEntityNode()
-    {
-        return $this->closeNode(self::ENTITY_NODE);
-    }
-
-    /**
-     * @param string $text
-     * @param array  $characterList
-     * @param array  $entities
-     *
-     * @return string
-     */
-    private function renderText($text = '', array $characterList = [], array $entities = [])
+    public function build(ContentState $contentState)
     {
         $output = '';
-        $stack = [];
-        $currentEntity = null;
-
-        $chars = str_split($text);
-
-        foreach ($chars as $index => $char) {
-            $characterMetadata = $characterList[$index];
-
-            $styles = $characterMetadata->getStyles();
-            $entityKey = $characterMetadata->getEntity();
 
-            $currentDepth = count($stack);
-
-            if ($entityKey !== $currentEntity || count($styles) !== $currentDepth) {
-                // close text node
-                if ($currentDepth > 0) {
-                    $output .= $this->closeTextNode();
-                }
+        $contentBlocks = $contentState->getBlockMap();
+        $entities = $contentState->getEntityMap();
 
-                // close link node
-                if ($entityKey !== $currentEntity && is_null($entityKey)) {
-                    $output .= $this->closeEntityNode();
-                }
+        $iterator = new \ArrayIterator($contentBlocks);
+        $iterator->rewind();
 
-                // create link node
-                if ($entityKey !== $currentEntity && !is_null($entityKey)) {
-                    $entity = $entities[$entityKey];
-                    $output .= $this->openEntityNode($entity);
-                }
+        while ($iterator->valid()) {
+            $renderer = $this->blockGuesser->getRenderer($iterator->current());
 
-                // create text node
-                if (count($styles) > 0) {
-                    $output .= $this->openTextNode($styles);
-                }
+            if (!$renderer) {
+                throw new DraftjsException(
+                    sprintf('Undefined block renderer for type "%s"', $iterator->current()->getType())
+                );
             }
 
-            $output .= $char;
-
-            $stack = $styles;
-            $currentEntity = $entityKey;
+            $output .= $renderer->render($iterator, $entities);
```

Comment:
Il serait très intéressant de lancer une exception si l'itérateur n'a pas avancé à la suite du `render`